pkg/kube/wait.go

```diff
@@ -68,7 +69,12 @@ func (w *waiter) waitForResources(created ResourceList) error {
 	ctx, cancel := context.WithTimeout(context.Background(), w.timeout)
 	defer cancel()
 
-	return wait.PollImmediateUntil(2*time.Second, func() (bool, error) {
+	timeout := 2 * time.Second
+	if w.perResourceTimeout > timeout {
+		timeout = w.perResourceTimeout
+	}
+
+	return wait.PollImmediateUntil(timeout, func() (bool, error) {
```

Comment:
As I already stated in https://github.com/helm/helm/issues/11535#issuecomment-1313980742, the first argument to `PollImmediateUntil` is **NOT the timeout** it is the polling interval.

See documentation:


> `func PollImmediateUntil(interval time.Duration, condition ConditionFunc, stopCh <-chan struct{}) error`
> PollImmediateUntil tries a condition func until it returns true, an error or stopCh is closed.
> PollImmediateUntil runs the 'condition' before waiting for the interval. 'condition' will always be invoked at least once.